### DISCUSSION TITLE
Where to put PATH and environment variable setup other than .zprofile for zsh?

### DISCUSSION THREAD

**Discussion by rdong8** [pyenv/pyenv#2200] [Q&A]
Putting the following lines
```
export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init --path)"
```
in .zprofile does not do anything for me for some reason. Is there somewhere else I can put them?

**Comment by native-api** [comment]
Assuming you're on Mac, check if you run Zsh as a login shell.
https://github.com/pyenv/pyenv/wiki/MacOS-login-shell

**Reply by rdong8** [reply]
I'm on linux. I see that there's an option in gnome-terminal to run as a login shell, is this the best way to do it?

**Reply by native-api** [reply]
If you're in Linux GUI, put those into `~/.profile` instead (unless you wish to connect via SSH, too, then to both profile and zprofile) and relogin.

**Reply by rdong8** [reply]
Thank you! This worked.
